src/eval/operation.rs

```diff
@@ -532,13 +533,19 @@ fn process_unary_operation(
                         let ts = ts
                             .into_iter()
                             .map(|t| {
-                                RichTerm::new(Term::App(f_as_var.clone(), t), pos_op_inh)
+                                let t_with_ctrs = apply_contracts(
+                                    t,
+                                    attrs.pending_contracts.iter().cloned(),
+                                    pos.into_inherited(),
+                                );
+
+                                RichTerm::new(Term::App(f_as_var.clone(), t_with_ctrs), pos_op_inh)
                                     .closurize(&mut shared_env, env.clone())
                             })
                             .collect();
 
                         Ok(Closure {
-                            body: RichTerm::new(Term::Array(ts, attrs.closurized()), pos_op_inh),
+                            body: RichTerm::new(Term::Array(ts, ArrayAttrs::new().closurized()), pos_op_inh),
```

Comment:
I may overthinking, but if we add new attributes to `ArrayAttrs`, this may not do what we want. Maybe something like `attrs.contracts_cleared().closurized()`, or something? Or just an explicit constructor (that would cause a compile error if `ArrayAttrs` is changed)?

Reply:
I went ahead and used `contracts_cleared` as it fits well with the existing methods.